gpii/node_modules/contextManager/src/ContextManager.js

```diff
@@ -160,16 +160,18 @@ gpii.contextManager.stopTemporalEnvironmentReporter = function (that) {
  *    active context or no user is logged in, undefined will be returned.
  **/
 gpii.contextManager.contextChanged = function (that, lifecycleManager, forceContextName) {
-    // find logged in users
-    var activeSessions = lifecycleManager.getActiveSessionGpiiKeys();
-    // if noone is logged in, do nothing
-    if (activeSessions.length === 0) {
+    var activeGpiiKeys = lifecycleManager.getActiveSessionGpiiKeys();
+
+    // if no one is logged in, or, "noUser" or "restore" is logged in, do nothing
+    if (activeGpiiKeys.length === 0 || activeGpiiKeys[0] === "noUser" || activeGpiiKeys[0] === "restore") {
```

Comment:
This should become a centralised utility - we can't expect every client to be able to keep track of all the strange conditions on whether a session is a genuine one. I expect if you hunt for "noUser" etc. you'll find a couple of other sites that could already be consolidated with this logic